cpp/include/raft/linalg/strided_reduction.cuh

```diff
@@ -71,6 +74,93 @@ void stridedReduction(OutType* dots,
   detail::stridedReduction(dots, data, D, N, init, stream, inplace, main_op, reduce_op, final_op);
 }
 
+/**
+ * @defgroup strided_reduction Strided Memory Access Reductions
+ * For reducing along rows for row-major and along columns for column-major
+ * @{
+ */
+
+/**
+ * @brief Compute reduction of the input matrix along the strided dimension
+ *        This API is to be used when the desired reduction is NOT along the dimension
+ *        of the memory layout. For example, a row-major matrix will be reduced
+ *        along the rows whereas a column-major matrix will be reduced along
+ *        the columns.
+ *
+ * @tparam InElementType the input data-type of underlying raft::matrix_view
+ * @tparam LayoutPolicy The layout of Input/Output (row or col major)
+ * @tparam OutElementType the output data-type of underlying raft::matrix_view and reduction
+ * @tparam IndexType Integer type used to for addressing
+ * @tparam MainLambda Unary lambda applied while acculumation (eg: L1 or L2 norm)
+ * It must be a 'callable' supporting the following input and output:
+ * <pre>OutType (*MainLambda)(InType, IdxType);</pre>
+ * @tparam ReduceLambda Binary lambda applied for reduction (eg: addition(+) for L2 norm)
+ * It must be a 'callable' supporting the following input and output:
+ * <pre>OutType (*ReduceLambda)(OutType);</pre>
+ * @tparam FinalLambda the final lambda applied before STG (eg: Sqrt for L2 norm)
+ * It must be a 'callable' supporting the following input and output:
+ * <pre>OutType (*FinalLambda)(OutType);</pre>
+ * @param[in] handle raft::handle_t
+ * @param[in] data Input of type raft::device_matrix_view
+ * @param[out] dots Output of type raft::device_matrix_view
+ * @param[in] init initial value to use for the reduction
+ * @param[in] main_op fused elementwise operation to apply before reduction
+ * @param[in] reduce_op fused binary reduction operation
+ * @param[in] final_op fused elementwise operation to apply before storing results
+ * @param[in] inplace reduction result added inplace or overwrites old values?
```

Comment:
It's unclear to me from reading this doc 1) whether or not the `dots` output is needed when `inplace==true` and 2) if `data == dots` when `inplace==true`. I felt the same while looking over the API for `reduce`.  This might just require some additional documentation in these functions, though I wonder if it's an indication that an update to the API might be needed (I'm just really not sure what that would be as of yet).

Reply:
It's unclear to me as well, and I mentioned it to another of @mhoemmen 's comments. Reading the impl did not make it any clearer for me.

Reply:
Maybe we should do a separate overload for the inplace version like we do in `raft::matrix`. That makes it much more straightforward to the user that it's in place (and which array(s) end up being overwriten).